code/datums/helper_datums/teleport.dm

```diff
@@ -189,6 +192,22 @@
 				if(O.opacity)
 					boominess += 10
 
+			if(istype(teleatom, /obj/mecha))
+				valid = 1
+				var/obj/mecha/M = teleatom
+				if(istype(M, /obj/mecha))

```

Comment:
You don't need to do this check, as M should be of mecha type since teleatom is.